src/server/game/Storages/WhoListStorage.cpp

```diff
@@ -0,0 +1,65 @@
+/*
+* Copyright (C) 2008-2016 TrinityCore <http://www.trinitycore.org/>
+*
+* This program is free software; you can redistribute it and/or modify it
+* under the terms of the GNU General Public License as published by the
+* Free Software Foundation; either version 2 of the License, or (at your
+* option) any later version.
+*
+* This program is distributed in the hope that it will be useful, but WITHOUT
+* ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+* FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+* more details.
+*
+* You should have received a copy of the GNU General Public License along
+* with this program. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include "WhoListStorage.h"
+#include "World.h"
+#include "ObjectAccessor.h"
+#include "Player.h"
+#include "GuildMgr.h"
+#include "WorldSession.h"
+
+WhoListStorageMgr* WhoListStorageMgr::instance()
+{
+    static WhoListStorageMgr instance;
+    return &instance;
+}
+
+void WhoListStorageMgr::Update()
+{
+    // clear current list
+    _whoListStorage.clear();
+    _whoListStorage.reserve(sWorld->GetPlayerCount()+1);
+
+    boost::shared_lock<boost::shared_mutex> lock(*HashMapHolder<Player>::GetLock());
```

Comment:
True, it's unneeded because this is done in main loop and even if other thread could run parallel to this it should not modify the player storage (except for reading)